### PR TITLE
Initial scaffolding of GHA work

### PR DESCRIPTION
**Pull request by stack72**
Fixes: #572

### CODE REVIEW

**Comment by stack72** [comment]
I am not going to merge this until after the long holiday weekend

File: nodejs/examples/metrics/mixed/package.json [hunk 1]
```diff
@@ -8,8 +8,8 @@
         "build": "tsc"
     },
     "dependencies": {
-        "@pulumi/pulumi": "^1.0.0",
-        "@pulumi/aws": "^2.0.0",
+        "@pulumi/pulumi": "^2.0.0",
```
Comment (leezen) [inline nodejs/examples/metrics/mixed/package.json:11]: We actually want this to stick to pulumi 1.x -- the point of this test is to check how the potential for SxS pulumi SDK versions works.

File: .github/workflows/master.yml [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+        run: make istanbul_tests
+    strategy:
+      fail-fast: true
+      matrix:
+        goversion:
+          - 1.14.x
+        language:
+          - nodejs
+        nodeversion:
+          - 13.x
```
Comment (leezen) [inline .github/workflows/master.yml:88]: Shall we bump this to 14?

File: Makefile [hunk 3]
```diff
@@ -1,20 +1,55 @@
...
+#.PHONY: check_clean_worktree
+#check_clean_worktree:
+#	$$(go env GOPATH)/src/github.com/pulumi/scripts/ci/check-worktree-is-clean.sh
+#
+## The travis_* targets are entrypoints for CI.
+#.PHONY: travis_cron travis_push travis_pull_request travis_api
+#travis_cron: all
+#travis_push: only_build check_clean_worktree only_test publish_packages
+#travis_pull_request: only_build check_clean_worktree only_test_fast
+#travis_api: all
```
Comment (leezen) [inline Makefile:55]: Why keep the commented out code?

**Review by leezen — COMMENTED** [review]
Given how much we repeat ourselves w/ the yaml, I assume this is generated with some tooling? In which case, is there some way to trace back how the file is generated?

File: Makefile (hunk 3 above)
Comment (stack72) [reply Makefile:55]: Gone!

File: .github/workflows/master.yml (hunk 2 above)
Comment (stack72) [reply .github/workflows/master.yml:88]: I was trying to make this like for like for now - I can follow up on this one though when we test for NodeJS 14 compatibility

File: nodejs/examples/metrics/mixed/package.json (hunk 1 above)
Comment (stack72) [reply nodejs/examples/metrics/mixed/package.json:11]: AH great spot! Thanks for that - fixing up now

**Comment by stack72** [comment]
@leezen I am going to address the YAML generation in a different move - the first thing was to get this working, then second will be to make the changes to the yaml generator in our ci-mgmt - that is the easy part :) But no changes will be visible for this repo

**Review by jaxxstorm — APPROVED** [review]

**Review by leezen — APPROVED** [review]

**Pull request merged**